controllers/service.go

```diff
@@ -182,6 +146,23 @@ func (r *SingleClusterReconciler) createSTSLoadBalancerSvc() error {
 		return err
 	}
 
+	if len(service.Spec.Ports) == 1 && service.Spec.Ports[0].Port == servicePort.Port &&
+		service.Spec.Ports[0].TargetPort == servicePort.TargetPort &&
+		service.Spec.Ports[0].Name == servicePort.Name {
+		return nil
+	}
+
+	service.Spec.Ports = []corev1.ServicePort{
+		servicePort,
+	}
+	if err := r.Client.Update(
+		context.TODO(), service, updateOption,
+	); err != nil {
+		return fmt.Errorf(
+			"failed to update service %s: %v", serviceName, err,
+		)
+	}
+
 	r.Log.Info(
 		"LoadBalancer Service already exist for cluster. Using existing service",
```

Comment:
Update Info msg saying updated. Also would be better to add similar msg saying 
`LoadBalancer Service already exist for cluster, trying to update if required"` before the update code flow